modules/docs/mdbook/package.nix

```diff
@@ -0,0 +1,33 @@
+{
+  mdbook,
+  mdbook-alerts,
+  runCommand,
+  writeTOML,
+  menu,
+  settings,
+  src,
+}:
+runCommand "html-docs"
+  {
+    inherit src;
+
+    nativeBuildInputs = [
+      mdbook
+      mdbook-alerts
+    ];
+
+    settings = writeTOML "book.toml" settings;
+    menu = builtins.toFile "menu.md" (builtins.unsafeDiscardStringContext menu);
```

Comment:
A comment would be nice to say what is discarded, I guess the why is that docs are "inert"?

Reply:
I copied this from the nixos docs impl, and there was no comment to copy.

However, yeah "string context" is how nix keeps track of a string's dependencies. E.g. if you do `"${pkgs.hello}/bin/hello"` then that string has some "context" so that nix knows it has a dependency on the `pkgs.hello` derivation.

Using `unsafeDiscardStringContext` drops that context, so `unsafeDiscardStringContext "${pkgs.hello}/bin/hello"` would have the same string value, but without the dependency on building the `pkgs.hello` derivation.

In the context of the docs, I believe this is mainly relevant for the "declared by" links, and maybe any `default` or `example` values that accidentally used a derivation in its value. I.e. we want the string value, but we don't want the dependency.